### PR TITLE
chore: get rid of openssl

### LINKED ISSUE #2200: Get rid of openssl
The only thing that prevents us from cross-compiling the whole codebase (instead of just client packages) is openssl (which is harder to get for cross-compilation targets).

Seems like:

https://github.com/bitcoindevkit/rust-esplora-client/pull/44

might fix it. I'm not sure if there's any other dependency that pulls it in.

### PR DESCRIPTION
**Pull request by jkitman**
Fixes #2200

### CODE REVIEW

File: fedimint-core/src/api.rs [hunk 1]
```diff
@@ -692,7 +691,6 @@ impl JsonRpcClient for WsClient {
     async fn connect(url: &Url) -> result::Result<Self, JsonRpcError> {
         #[cfg(not(target_family = "wasm"))]
         return WsClientBuilder::default()
-            .certificate_store(CertificateStore::WebPki)
```
Comment (jkitman) [inline fedimint-core/src/api.rs:695]: I guess this can safely be removed?

**Comment by jkitman** [comment]
Fails during `ecash_can_be_recovered` when `client.request` never returns (server works fine).

File: fedimint-core/Cargo.toml [hunk 2]
```diff
@@ -19,8 +19,9 @@ futures = "0.3.24"
 bincode = "1.3.1"
 bech32 = "0.9.1"
 itertools = "0.10.5"
-jsonrpsee-types = "0.16.0"
-jsonrpsee-core = { version = "0.16.2", features = [ "client" ] }
+# TODO: use official release, but right now there's a bug that stalls our client
+jsonrpsee-types = { version = "0.18.0" }
```
Comment (justinmoon) [inline fedimint-core/Cargo.toml:23]: I don't think this comment is relevant anymore?

File: fedimint-client-legacy/Cargo.toml [hunk 3]
```diff
@@ -45,17 +45,18 @@ tbs = { path = "../crypto/tbs" }
 thiserror = "1.0.39"
 tokio = { version = "1.26.0", features = ["sync"] }
 tracing = "0.1.37"
-jsonrpsee-types = "0.16.0"
-jsonrpsee-core = { version = "0.16.2", features = [ "client" ] }
+# TODO: use official release, but right now there's a bug that stalls our client
+jsonrpsee-types = "0.18.0"
```
Comment (justinmoon) [inline fedimint-client-legacy/Cargo.toml:49]: can remove this comment

**Review by justinmoon — APPROVED** [review]

File: misc/git-hooks/pre-commit [hunk 4]
```diff
@@ -54,6 +54,12 @@ function check_forbidden_dependencies() {
       return 1
     fi
   done
+  find ./ -name Cargo.lock | while read -r cargo_lock ; do
```
Comment (dpc) [inline misc/git-hooks/pre-commit:57]: :heart:

**Pull request merged**